### PR TITLE
Scheduler test

### PR DESCRIPTION
**Pull request by razvan**
ZooKeeper integration test for pod re-scheduling

### CODE REVIEW

File: tests/scheduler.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+use anyhow::Result;
+use common::zookeeper::{build_test_cluster, build_zk_cluster};
+use stackable_zookeeper_crd::ZookeeperVersion;
+
+/// Tests that pods are rescheduled on the same nodes.
+/// Requirements: At least 4 nodes.
+/// How it works:
+/// 1. creates a cluster with 3 replicas and saves the selected node names.
+/// 2. deletes all pods by creating a cluster with 0 replicas
+/// 3. recreates the 3 replicas again and compares the selected node names with the ones from step 1.
```
Comment (maltesander) [inline tests/scheduler.rs:13]: In the readme we write of a dynamic amount of nodes (well tested with 3) and @backstreetkiwi uses 3 as well with T2. Can you adapt the docs and notifiy him?

File: tests/scheduler.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        .iter()
+        .map(|p| {
+            p.spec
+                .as_ref()
+                .map_or("".to_string(), |s| s.node_name.as_ref().unwrap().clone())
+        })
+        .collect();
+
+    // println!("node_names_1: {:?}", node_names_1);
+    // println!("node_names_2: {:?}", node_names_2);
```
Comment (maltesander) [inline tests/scheduler.rs:61]: leftovers?

**Review by maltesander — CHANGES_REQUESTED** [review]
Just 2 minor things.

File: tests/scheduler.rs (hunk 1 above)
Comment (razvan) [reply tests/scheduler.rs]: I reduced the required no. of nodes to two and only test with one pod.

File: tests/scheduler.rs (hunk 2 above)
Comment (razvan) [reply tests/scheduler.rs]: Removed,

**Review by backstreetkiwi — APPROVED** [review]
I blackbox-tested this PR in Jenkins: https://ci.stackable.tech/job/Zookeeper%20Operator%20Integration%20Tests/8/

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request merged**